From d44c1fe70d30aef3e7cebd3af84e2f71fae811ab Mon Sep 17 00:00:00 2001
From: Jack Baron <jackmbaron@gmail.com>
Date: Tue, 12 Apr 2022 02:21:30 +0100
Subject: [PATCH 1/4] feat: add repo arg to dockerfile

---
 Dockerfile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Dockerfile b/Dockerfile
index 966086c5..cd9a990b 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -24,5 +24,8 @@ RUN addgroup -g 1001 -S nodejs && \
 
 USER nodejs
 
+ARG GIT_REPO
+LABEL org.opencontainers.image.source=${GIT_REPO}
+
 EXPOSE 3000
 CMD ["yarn", "start"]
-- 
2.34.1.windows.1

