From a457562895e3d66d5e36cfaa9a96b583ffad50c2 Mon Sep 17 00:00:00 2001
From: Jack Baron <jackmbaron@gmail.com>
Date: Sat, 17 Jul 2021 13:21:40 +0100
Subject: [PATCH] feat: add repo arg to dockerfile

---
 Dockerfile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Dockerfile b/Dockerfile
index da24d60a..255d09ac 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -46,5 +46,8 @@ RUN addgroup -g 1001 -S nodejs && \
 
 USER nodejs
 
+ARG GIT_REPO
+LABEL org.opencontainers.image.source=${GIT_REPO}
+
 EXPOSE 3000
 CMD ["yarn", "start"]
-- 
2.30.1.windows.1

