From 4f7410e44e99ecb3243095490896d1eb9ae4453c Mon Sep 17 00:00:00 2001
From: Jack Baron <jackmbaron@gmail.com>
Date: Sat, 17 Jul 2021 13:32:39 +0100
Subject: [PATCH] optimise dockerfile

---
 Dockerfile | 49 +++++++++++++++++++++++++++++++++++++------------
 1 file changed, 37 insertions(+), 12 deletions(-)

diff --git a/Dockerfile b/Dockerfile
index 63dee906..b00acbea 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,26 +1,51 @@
-FROM node:14-alpine
-
-ENV APP_PATH /opt/outline
-RUN mkdir -p $APP_PATH
+# syntax=docker/dockerfile:1.2
+FROM node:14-alpine AS deps-common
 
+ARG APP_PATH /opt/outline
 WORKDIR $APP_PATH
+COPY ./package.json ./yarn.lock ./
+
+# ---
+FROM deps-common AS deps-dev
+RUN yarn install --no-optional --frozen-lockfile && \
+  yarn cache clean
 
-COPY package.json ./
-COPY yarn.lock ./
+# ---
+FROM deps-common AS deps-prod
+RUN yarn install --production=true --frozen-lockfile && \
+  yarn cache clean
 
-RUN yarn --pure-lockfile
+# ---
+FROM node:14-alpine AS builder
+WORKDIR $APP_PATH
 
 COPY . .
+COPY --from=deps-dev $APP_PATH/node_modules ./node_modules
+RUN yarn build
 
-RUN yarn build && \
-  yarn --production --ignore-scripts --prefer-offline && \
-  rm -rf shared && \
-  rm -rf app
+# ---
+FROM node:14-alpine AS runner
 
+WORKDIR $APP_PATH
 ENV NODE_ENV production
-CMD yarn start
+
+RUN apk add --no-cache tini
+
+COPY --from=builder $APP_PATH/build ./build
+COPY --from=builder $APP_PATH/server ./server
+COPY --from=builder $APP_PATH/.sequelizerc ./.sequelizerc
+COPY --from=deps-prod $APP_PATH/node_modules ./node_modules
+COPY --from=builder $APP_PATH/package.json ./package.json
+
+RUN addgroup -g 1001 -S nodejs && \
+  adduser -S nodejs -u 1001 && \
+  chown -R nodejs:nodejs $APP_PATH/build
+
+USER nodejs
 
 ARG GIT_REPO
 LABEL org.opencontainers.image.source=${GIT_REPO}
 
 EXPOSE 3000
+ENTRYPOINT ["/sbin/tini", "--"]
+CMD ["yarn", "start"]
-- 
2.30.1.windows.1


